File: applicatie/src/main/java/com/vdzon/administratie/bankimport/regelParsers/SepaRegelParser.java
package com.vdzon.administratie.bankimport.regelParsers;

import static com.vdzon.administratie.bankimport.ImportFromAbnAmroHelper.findNextKeyword;

public class SepaRegelParser implements RegelParser {

    //TODO: deze class kan wat mooier

    @Override
    public boolean match(String omschrijving) {
        return omschrijving.startsWith("SEPA");
    }


    @Override
    public String extractNaam(String omschrijving) {
        int pos = omschrijving.indexOf("Naam:");
        int posStart = pos + "Naam:".length();
        if (pos > 0) {
            String nextKeyword = findNextKeyword(omschrijving.substring(posStart));
            if (nextKeyword == null) {
                return omschrijving.substring(posStart);
            } else {
                int posEnd = posStart + omschrijving.substring(posStart).indexOf(nextKeyword);
                return omschrijving.substring(posStart, posEnd - 1);
            }
        }
        return "Onbekend";
    }

    @Override
    public String extractOmschrijving(String omschrijving) {
        int pos = omschrijving.indexOf("Omschrijving:");
        int posStart = pos + "Omschrijving:".length();
        if (pos > 0) {
            String nextKeyword = findNextKeyword(omschrijving.substring(posStart));
            if (nextKeyword == null) {
                return omschrijving.substring(posStart);
            } else {
                int posEnd = posStart + omschrijving.substring(posStart).indexOf(nextKeyword);
                return omschrijving.substring(posStart, posEnd - 1);
            }
        }
        return omschrijving;
    }
}

File: applicatie/src/main/java/com/vdzon/administratie/bankimport/regelParsers/SepaRegelParser.kt
package com.vdzon.administratie.bankimport.regelParsers

import com.vdzon.administratie.bankimport.ImportFromAbnAmroHelper.findNextKeyword

class SepaRegelParser : RegelParser {

    //TODO: deze class kan wat mooier

    override fun match(omschrijving: String): Boolean {
        return omschrijving.startsWith("SEPA")
    }

    override fun extractNaam(omschrijving: String): String {
        val pos = omschrijving.indexOf("Naam:")
        val posStart = pos + "Naam:".length
        if (pos > 0) {
            val nextKeyword = findNextKeyword(omschrijving.substring(posStart))
            if (nextKeyword == null) {
                return omschrijving.substring(posStart)
            } else {
                val posEnd = posStart + omschrijving.substring(posStart).indexOf(nextKeyword)
                return omschrijving.substring(posStart, posEnd - 1)
            }
        }
        return "Onbekend"
    }

    override fun extractOmschrijving(omschrijving: String): String {
        val pos = omschrijving.indexOf("Omschrijving:")
        val posStart = pos + "Omschrijving:".length
        if (pos > 0) {
            val nextKeyword = findNextKeyword(omschrijving.substring(posStart))
            if (nextKeyword == null) {
                return omschrijving.substring(posStart)
            } else {
                val posEnd = posStart + omschrijving.substring(posStart).indexOf(nextKeyword)
                return omschrijving.substring(posStart, posEnd - 1)
            }
        }
        return omschrijving
    }
}